File: apps/frontend/lib/src/features/maintenance_ticket/domain/entities/space_entity.dart
import 'package:equatable/equatable.dart';

class SpaceEntity extends Equatable {
  const SpaceEntity({
    required this.id,
    this.code,
    this.name,
    this.description,
    this.siteId,
    this.isActive = true,
  });

  final String id;
  final String? code;
  final String? name;
  final String? description;
  final String? siteId;
  final bool isActive;

  /// Display name for UI
  String get displayName {
    if (name != null && name!.isNotEmpty) {
      return name!;
    }
    if (code != null && code!.isNotEmpty) {
      return code!;
    }
    return 'Space $id';
  }

  @override
  List<Object?> get props => [
        id,
        code,
        name,
        description,
        siteId,
        isActive,
      ];
}

File: apps/backend/src/modules/tenant/entities/space.entity.ts
import {
  Column,
  Entity,
  Index,
  ManyToOne,
} from 'typeorm';
import { TenantBaseEntity } from '../../../shared/database/tenant-base.entity';
import { Site } from './site.entity';
import { SpaceCategory } from './space-category.entity';

@Entity({ name: 'spaces' })
@Index(['companyId', 'code'])
@Index(['companyId', 'siteId'])
export class Space extends TenantBaseEntity {
  @Column({ type: 'varchar', length: 100 })
  code!: string;

  @Column({ type: 'varchar', length: 255 })
  name!: string;

  @Column({ type: 'uuid', nullable: true, name: 'site_id' })
  siteId?: string;

  @ManyToOne(() => Site, { nullable: true })
  site?: Site;

  @Column({ type: 'uuid', nullable: true, name: 'space_category_id' })
  spaceCategoryId?: string;

  @ManyToOne(() => SpaceCategory, { nullable: true })
  spaceCategory?: SpaceCategory;

  @Column({ type: 'text', nullable: true })
  description?: string;

  @Column({ type: 'boolean', default: true, name: 'is_active' })
  isActive!: boolean;
}

